glyphs-reader/src/glyphdata_bundled.rs

```diff
@@ -86,31 +91,31 @@ fn name(idx: usize) -> &'static str {
 }
 
 pub(crate) fn find_pos_by_name(needle: &str) -> Option<usize> {
-    let mut upper = NAME_OFFSETS.len();
+    let mut upper = NAME_OFFSETS.len() as i32;
     let mut lower = 0;
     while lower <= upper {
-        let mid = (lower + upper) / 2;
+        let mid = ((lower + upper) / 2) as usize;
         let c = name(mid).cmp(needle);
         match c {
             Ordering::Equal => return Some(mid),
-            Ordering::Less => lower = mid + 1,
-            Ordering::Greater => upper = mid - 1,
+            Ordering::Less => lower = mid as i32 + 1,
+            Ordering::Greater => upper = mid as i32 - 1,
         }
     }
     None
 }
 
 pub(crate) fn find_pos_by_codepoint(needle: u32) -> Option<usize> {
-    let mut upper = CODEPOINT_TO_INFO_IDX.len();
+    let mut upper = CODEPOINT_TO_INFO_IDX.len() as i32;
     let mut lower = 0;
     while lower <= upper {
-        let mid = (lower + upper) / 2;
+        let mid = ((lower + upper) / 2) as usize;
         let (cp, i) = CODEPOINT_TO_INFO_IDX.get(mid).unwrap();
         let c = cp.cmp(&(needle as usize));
         match c {
             Ordering::Equal => return Some(i),
-            Ordering::Less => lower = mid + 1,
-            Ordering::Greater => upper = mid - 1,
+            Ordering::Less => lower = mid as i32 + 1,
```

Comment:
now that i'm looking at this with fresh eyes, is there some reason we don't want to use the built in binary search? It also seems like these two fns could be sharing mot of their code..

Reply:
Unless I'm missing something obvious the built-in is defined on Vec. We don't have a Vec. Code sharing is probably possible indeed, will add.

Reply:
the builtins are defined on slice, vec gets them through deref.

Reply:
It's indirect in some cases, e.g. array of offsets and you get two that define a string. Or a slice of 2-tuples where you are searching .0 and returning .1. I've generalized it to a single bsearch method which is hopefully a good start, we can iterate further over time.